.github/workflows/needs_release_notes.yml

```diff
@@ -11,7 +11,12 @@ jobs:
       pull-requests: write
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/labeler@8558fd74291d67161a8a78ce36a881fa63b766a9  # v5.0.0
+    - uses: actions/checkout@v4
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
Why do we need to checkout the repo?

Reply:
Failure to find `.github/labeler.yml`, which was the first error being encountered on #2533 and led me to looking into this